### ISSUE TITLE
Flash Attention Support

### ISSUE THREAD

**Issue by vgoklani** [NVIDIA/TransformerEngine#183]
Hey there,

Does this library support Flash Attention (Hazy Research)? Are there any constraints on either the head-dimension or sequence length? What about the PyTorch 2.0 SDPA implementation? We are mainly interested in using FP8 via Flash Attention. Also want to add a negative bias prior to softmax (ALiBi).

What happens if we try to use the PyTorch 2.0 compiler with this framework? Is everything fully supported? Will the compiler automatically fuse the various layers? Or should we just follow the tutorial and fuse everything manually?

One more thing, we have a pre-trained model that was trained using FP32. We would like to extend the pre-training process. Is it possible to first convert that model to FP8, and then continue training in FP8? Do you have a process we can follow for doing performing that down-scaling step?

Thanks!

**Comment by SinanAkkoyun** [comment]
Hi, why was this closed?
